### ISSUE TITLE
External pressure may lead to computational instability

### ISSUE THREAD

**Issue by scott-5** [deepmodeling/abacus-develop#2699]
### Describe the bug

This example can be calculated normally without the press* parameter, but with the press* parameter, errors are reported like that:

```
abacus: /home/user/abacus-develop/source/module_cell/module_symmetry/symmetry.cpp:152: void ModuleSymmetry::Symmetry::analy_sys(const UnitCell &, std::basic_ofstream<char, std::char_traits<char>> &): Assertion `tmp_nrotk == this->nrotk' failed.
```

### To Reproduce

It can be reproduced by uploaded files.
[example.zip](https://github.com/deepmodeling/abacus-develop/files/11960193/example.zip)

### Environment

ABACUS version is 3.2.5.

### Additional Context

Since the structure can be calculated in VASP regardless of whether there is external pressure, the initial structure should be reasonable.

**Comment by scott-5** [comment]
Similarly other structures can still be calculated normally. It just means that the tolerance for poorer initial structures may need to be increased.

**Comment by maki49** [comment]
The bug results from not taking into account the case where symmetry increases during cell-relax. 
It will be fixed in #2700.

**Comment by scott-5** [comment]
It seems like a similar problem after fixing the bug.
```
abacus: /home/user/abacus-develop/source/module_cell/module_symmetry/symmetry.cpp:157: void ModuleSymmetry::Symmetry::analy_sys(const UnitCell &, std::basic_ofstream<char, std::char_traits<char>> &): Assertion `tmp_nrotk == this->nrotk' failed.
```
[example2.zip](https://github.com/deepmodeling/abacus-develop/files/11995713/example2.zip)

**Comment by maki49** [comment]
@scott-5 The bug will be fixed in #2713. 
Thanks for providing such an interesting case that, in a certain ion-step: 
the symmetry firstly becomes poorer than the last step, which forces the program to enlarge `symmetry_prec` and analyze again, then the reanalyse gives a symmetry result better than the last step.

**Comment by scott-5** [comment]
Not at all. Such a changing situation is indeed interesting.
